### PR TITLE
[DBG] Fix compilation warnings

### PR DESCRIPTION
**Pull request by smuzaffar**
Remove explicit `#define EDM_ML_DEBUG` defines which are not only generating compiltion warnings for DBG IBs ( where `EDM_ML_DEBUG`  is set by `scram`) but it also prints the edm debug messages for production.

### CODE REVIEW

**Comment by smuzaffar** [comment]
please test

File: Geometry/HGCalTBCommonData/plugins/HGCalTBNumberingInitialization.cc [hunk 1]
```diff
@@ -26,8 +26,6 @@
 #include "Geometry/HGCalTBCommonData/interface/HGCalTBParameters.h"
 #include "Geometry/Records/interface/IdealGeometryRecord.h"
 
-#define EDM_ML_DEBUG
```
Comment (civanch) [inline Geometry/HGCalTBCommonData/plugins/HGCalTBNumberingInitialization.cc:29]: @smuzaffar , an alternative fix is not to remove this line but to comment it out.

Comment (civanch) [reply Geometry/HGCalTBCommonData/plugins/HGCalTBNumberingInitialization.cc:29]: In this class define and ifdef may be removed.

File: Geometry/HGCalTBCommonData/plugins/HGCalTBParametersESModule.cc [hunk 2]
```diff
@@ -11,8 +11,6 @@
 #include "Geometry/HGCalTBCommonData/interface/HGCalTBParametersFromDD.h"
 #include "Geometry/Records/interface/IdealGeometryRecord.h"
 
-#define EDM_ML_DEBUG
```
Comment (civanch) [inline Geometry/HGCalTBCommonData/plugins/HGCalTBParametersESModule.cc:14]: In this class in run time debug printout may be needed, so better to comment line 14 and not remove.

Comment (smuzaffar) [reply Geometry/HGCalTBCommonData/plugins/HGCalTBParametersESModule.cc:14]: @civanch , should I also remove `#ifdef ... #endif` around the `edm::LogVerbatim(..)` statements in this class?

Comment (civanch) [reply Geometry/HGCalTBCommonData/plugins/HGCalTBParametersESModule.cc:14]: @smuzaffar , in this class debugging may be useful, so I would simply comment out line 14 and not remove it.

File: Geometry/HGCalTBCommonData/plugins/HGCalTBNumberingInitialization.cc (hunk 1 above)
Comment (smuzaffar) [reply Geometry/HGCalTBCommonData/plugins/HGCalTBNumberingInitialization.cc:29]: done

File: Geometry/HGCalTBCommonData/plugins/HGCalTBParametersESModule.cc (hunk 2 above)
Comment (smuzaffar) [reply Geometry/HGCalTBCommonData/plugins/HGCalTBParametersESModule.cc:14]: done

**Comment by smuzaffar** [comment]
please test

**Comment by civanch** [comment]
+1

**Comment by perrotta** [comment]
+1

**Pull request merged**